cloudinit/distros/package_management/apt.py

```diff
@@ -123,19 +127,29 @@ def get_all_packages(self):
         return set(resp.splitlines())
 
     def get_unavailable_packages(self, pkglist: Iterable[str]):
-        return [pkg for pkg in pkglist if pkg not in self.get_all_packages()]
+        # Packages ending with `-` signify to apt to not install a transitive
+        # dependency.
+        # Anything after "/" refers to a target release
+        # "=" allows specifying a specific version
+        # Strip all off when checking for availability
+        return [
+            pkg
+            for pkg in pkglist
+            if re.split("/|=|-", pkg)[0] not in self.get_all_packages()
```

Comment:
The issue you'll have with spitting on hyphen is that many packages contain hyphens, like our own `cloud-init` :) which will lead to us trying to match `cloud` instead of cloud-init. It'll have to specifically be an rstrip for the hyphen case... or making sure we only truncate that last character if it's a hyphen.

Reply:
I'm +1 on splitting on the `/` or `=` as they can't be part of a [valid debian package name](https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-source). But again, the hyphen we need to treat differently once we have that split.
So maybe the following?
```suggestion
            if re.split("/|=", pkg)[0].rstrip("-") not in self.get_all_packages()
```